terraform/modules/kubernetes/kubernetes.tf

```diff
@@ -298,6 +321,146 @@ resource "kubernetes_cron_job" "workflow_manager" {
   }
 }
 
+resource "kubernetes_deployment" "intake_batch" {
+  metadata {
+    name      = "intake-batch-${var.ingestor}"
+    namespace = var.kubernetes_namespace
+  }
+
+  spec {
+    replicas = var.intake_batch_replica_count
+    selector {
+      match_labels = {
+        name = "intake-batch-${var.ingestor}"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          name = "intake-batch-${var.ingestor}"
+        }
+      }
+      spec {
+        service_account_name = module.account_mapping.kubernetes_account_name
+        container {
+          name  = "facile-container"
+          image = "${var.container_registry}/${var.facilitator_image}:${var.facilitator_version}"
+          args  = ["intake-batch-worker"]
+          resources {
+            # Batch intake is single-threaded, so we request an entire CPU for
+            # maximum throughput. We never expect to see batches larger than
+            # 3-400 MB, and batches are processed in memory, so request enough
+            # for whole ingestion batches plus a safety margin.
+            requests {
```

Comment:
Did you check what the used resources on small batches is? 500Mi and 1.0 CPU seems quite high imho. I'd start lower (but I don't know rust and how memory hungry it is). However, for CPU I'd go lower since you can bin pack more workers on one node this way and they'll use what they can get, if they can't get more things will just take a bit longer.

Reply:
The real batches we've seen thus far are much, much smaller than this. On average, about 3 KB for Apple and 135 KB for Google across MD and CT. You're right that since we no longer rely on the container resource requests to limit how many jobs run at once, we can set these much lower. I'll set 0.1 CPU and 50 MB of memory.